src/pages/storage/StorageVolumes.tsx

```diff
@@ -0,0 +1,150 @@
+import React, { FC, useState } from "react";
+import { useParams } from "react-router-dom";
+import { useQuery } from "@tanstack/react-query";
+import { queryKeys } from "util/queryKeys";
+import { MainTable, SearchBox, useNotify } from "@canonical/react-components";
+import Loader from "components/Loader";
+import { fetchStorageVolumes } from "api/storage-pools";
+import { isoTimeToString } from "util/helpers";
+import DeleteStorageVolumeBtn from "pages/storage/actions/DeleteStorageVolumeBtn";
+
+const StorageVolumes: FC = () => {
+  const notify = useNotify();
+  const [query, setQuery] = useState("");
+  const { name: pool, project } = useParams<{
+    name: string;
+    project: string;
+  }>();
+
+  if (!pool) {
+    return <>Missing storage pool name</>;
+  }
+  if (!project) {
+    return <>Missing project</>;
+  }
+
+  const {
+    data: volumes,
+    error,
+    isLoading,
+  } = useQuery({
+    queryKey: [queryKeys.storage, pool, queryKeys.volumes],
+    queryFn: () => fetchStorageVolumes(pool),
+  });
+
+  if (error) {
+    notify.failure("Loading storage volumes failed", error);
+  }
+
+  if (isLoading) {
+    return <Loader text="Loading storage volumes..." />;
+  }
+  if (!volumes) {
+    return <>Loading storage volumes failed</>;
+  }
+
+  const headers = [
+    { content: "Name", sortKey: "name" },
+    { content: "Content type", sortKey: "contentType" },
+    { content: "Type", sortKey: "type" },
+    { content: "Created at", sortKey: "createdAt" },
+    { content: "Location", sortKey: "location" },
+    { content: "Config" },
+    { content: "Used by" },
+    {
+      content: "",
+      "aria-label": "Actions",
+    },
+  ];
+
+  const filteredVolumes = volumes.filter((volume) => {
+    return volume.name.toLowerCase().includes(query.toLowerCase());
+  });
+
+  const rows = filteredVolumes.map((volume) => {
+    return {
+      columns: [
+        {
+          content: volume.name,
+          role: "rowheader",
+          "aria-label": "Name",
+        },
+        {
+          content: volume.content_type,
+          role: "rowheader",
+          "aria-label": "Content type",
+        },
+        {
+          content: volume.type,
+          role: "rowheader",
+          "aria-label": "Type",
+        },
+        {
+          content: isoTimeToString(volume.created_at),
+          role: "rowheader",
+          "aria-label": "Created at",
+        },
+        {
+          content: volume.location,
+          role: "rowheader",
+          "aria-label": "Location",
+        },
+        {
+          content: Object.entries(volume.config).map(([key, val]) => (
+            <div key={key}>
+              {key}: {val}
+            </div>
+          )),
+          role: "rowheader",
+          "aria-label": "Config",
+        },
+        {
+          content: volume.used_by?.map((entry) => (
+            <div key={entry}>{entry}</div>
```

Comment:
This doesn't look very well on medium/small screens. We should allow it to either wrap `anywhere` or add the `u-truncate` class and set the `title`.

![image](https://github.com/canonical/lxd-ui/assets/56583786/cc918d52-bb7f-4766-a4c8-af6d78d1066b)